src/base/iconprovider.cpp

```diff
@@ -57,7 +72,19 @@ IconProvider *IconProvider::instance()
 
 QString IconProvider::getIconPath(const QString &iconId) const
 {
-    return ":/icons/qbt-theme/" + iconId + ".png";
+    const auto it = m_iconFileNames.find(iconId);
+    if (it != m_iconFileNames.end()) {
+        return it.value();
+    }
+
+    const QString fileSuffix = QLatin1String("256/") + iconId + QLatin1String(".png");
+    for (const auto d: themeDirs) {
```

Comment:
Please use some meaningful name instead of `d`(e.g. themeDir).
And add missing space before `:`.